extensions/nanoarrow_ipc/src/nanoarrow/nanoarrow_ipc_decoder.c

```diff
@@ -1550,7 +1560,7 @@ static int ArrowIpcDecoderWalkSetArrayView(struct ArrowIpcArraySetter* setter,
     }
 
     setter->src.data_type = array_view->layout.buffer_data_type[i];
-    setter->src.element_size_bits = array_view->layout.element_size_bits[i];
+    setter->src.element_size_bits = (int32_t)array_view->layout.element_size_bits[i];
```

Comment:
Should we just reduce `element_size_bits` in `ArrowLayout`? It seems the chances that an element of an array can be so big that its size needs to be represented in 64bits are pretty low.